src/core/timer.c

```diff
@@ -37,7 +37,6 @@ nni_timer_sys_init(void)
 	int        rv;
 	nni_timer *timer = &nni_global_timer;
 
-	memset(timer, 0, sizeof(*timer));
```

Comment:
I think this one is still necessary.  We don't clear t_waiting in the section below, and there is a very low chance that a user can call nng_fini() causing a teardown, and then subsequently cause a new initialization.  This is not on any hot code path (library initialization only), so removing it doesn't really save anything.

Reply:
Hi~nni_global_timer

Reply:
Hi~
The **nni_global_timer** is a static variable, so it will be zero by default isn't it?

Reply:
After a few days finally I got what you mean, the library is capable of closing/opening multiple times. So the `memset` in `nni_timer_sys_init` is still necessary. A new commit had already been pushed.